src/nxt_runtime.c

```diff
@@ -797,6 +802,60 @@ nxt_runtime_conf_init(nxt_task_t *task, nxt_runtime_t *rt)
     return NXT_OK;
 }
 
+static nxt_int_t
+nxt_runtime_creds(nxt_task_t *task,  nxt_runtime_t *rt)
+{
+    char          *user, *group_name;
+    struct group  *grp;
+    struct passwd *pwd;
+
+    if (rt->capabilities.setid) {
+        if (rt->user_cred.user == NULL) {
+            rt->user_cred.user = NXT_USER;
+        }
+
+        if (rt->group == NULL) {
+            rt->group = NXT_GROUP;
+        }
+    } else {
+        pwd = getpwuid(geteuid());
+
+        if (pwd == NULL) {
+            user = group_name = NULL;
+        } else {
+            user = pwd->pw_name;
+            grp = getgrgid(pwd->pw_gid);
+
+            if (grp != NULL) {
+                group_name = grp->gr_name;
+            } else {
+                group_name = NULL;
+            }
+        }
+
+        if (user == NULL) {
+            nxt_alert(task, "Unit is unable to get the current username. "
+                "There's no entry for uid %d in passwd",
```

Comment:
I believe there can be more reasons for `getpwuid()` to fail... we should check `nxt_errno` for exact reason.

Reply:
from man:

```
0 or ENOENT or ESRCH or EBADF or EPERM or ...
              The given name or uid was not found.
```
Maybe then I can just use:

```
        if (user == NULL) {
            nxt_alert(task, "Unit is unable to get the current username %E", nxt_errno);
            return NXT_ERROR;
        }
```
?

Reply:
You can use similar code as here: https://github.com/nginx/unit/blob/master/src/nxt_process.c#L405